jax/experimental/checkify.py

```diff
@@ -495,16 +564,65 @@ def assert_discharge_rule(error, enabled_errors, pred, code, *, msgs):
 
 ## checkify api
 
-ErrorCategory = enum.Enum('ErrorCategory', ['NAN', 'OOB', 'DIV', 'ASSERT'])
+ErrorCategory = enum.Enum('ErrorCategory', ['NAN', 'OOB', 'DIV', 'USER_CHECK'])
 
-float_errors = {ErrorCategory.NAN, ErrorCategory.DIV}
-index_errors = {ErrorCategory.OOB}
+float_errors = frozenset({ErrorCategory.NAN, ErrorCategory.DIV})
+index_errors = frozenset({ErrorCategory.OOB})
 automatic_errors = float_errors | index_errors
-user_asserts = {ErrorCategory.ASSERT}
+user_checks = frozenset({ErrorCategory.USER_CHECK})
 
 Out = TypeVar('Out')
-def checkify(fun: Callable[..., Out], errors: Set[ErrorCategory] = user_asserts
+
+
+def checkify(fun: Callable[..., Out],
+             errors: FrozenSet[ErrorCategory] = user_checks
```

Comment:
```suggestion
             errors: Set[ErrorCategory] = user_checks
```

Reply:
QQ: I made this a FrozenSet because of a "dangerous default value" warning, but I guess the Set type is a super-set of the FrozenSet type, and it can still accept Sets even though the default value will be a FrozenSet? (I think I answered my own question)

Reply:
ah whoops, flake is not happy with that: `Incompatible default for argument "errors" (default has type "FrozenSet[ErrorCategory]", argument has type "Set[ErrorCategory]") `